sdk/core/Azure.Core/src/Http/DateConditionalRequestOptions.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+
+namespace Azure.Core.Http
+{
+    public class DateConditionalRequestOptions : ConditionalRequestOptions
+    {
+        /// <summary>
+        /// Optionally limit requests to resources that have only been
+        /// modified since this point in time.
+        /// </summary>
+        public DateTimeOffset? IfModifiedSince { get; set; }
+
+        /// <summary>
+        /// Optionally limit requests to resources that have remained
+        /// unmodified
+        /// </summary>
+        public DateTimeOffset? IfUnmodifiedSince { get; set; }
+
+        /// <summary>
+        /// Set preconditions that indicate to apply an operation only to
+        /// resources that have been modified since the specified date.
+        /// </summary>
+        public virtual void SetIfModifiedSinceCondition(DateTimeOffset dateTime)
+        {
+            IfModifiedSince = dateTime;
+        }
+
+        /// <summary>
+        /// Set preconditions that indicate to apply an operation only to
+        /// resources that have not been modified since the specified date.
+        /// </summary>
+        public virtual void SetIfUnmodifiedSinceCondition(DateTimeOffset dateTime)
```

Comment:
Not sure if these add a ton of value.

Reply:
You mean because the methods are redundant with just setting the properties directly?

Reply:
Yep, considering they have the same name.